testsuite/tests/typing-modules/mixmod5.ml

```diff
@@ -0,0 +1,334 @@
+          val eval : (string * exp) list -> exp0 -> exp
+        end
+      val map : (E.exp0 -> E.exp0) -> exp0 -> exp
+      val eval : (string * LVar.exp) list -> exp0 -> LVar.exp
+    end
+module type ExprF =
+  sig
+    type exp0 = a ExprT.exp
+    and a = < exp : exp0 >
+    type exp = exp0
+    val eval : (string * exp) list -> exp -> exp
+  end
+module rec ExprF : ExprF/2
+val e2 : ExprF.exp = `Add (`Num 6, `Var "x")
+|}]
+
+(* The lexpr language, reunion of lambda and expr *)
+
+module LExprT = struct
+  (* We don't need to write the constraint here, as we don't use
+     'e specifically *)
+  type 'a exp = [ 'a LamT.exp | 'a ExprT.exp ]
+end
+module type LExprS = sig
+  type exp0 = private [> a LExprT.exp]
+  and a = <exp:exp0>
+  include E with type exp = exp0
+end
+module LExpr(E : LExprS) =
+  struct
+    include Types(E)
+    type exp0 = E.a LExprT.exp
+    module SLam = Lam(E)
+    module SExpr = Expr(E)
+
+    let eval subst : exp0 -> exp = function
+        #SLam.exp0 as x -> SLam.eval subst x
+      | #SExpr.exp0 as x -> SExpr.eval subst x
+  end
+
+module type LExprF = sig
+  type exp0 = a LExprT.exp
+  and a = <exp:exp0>
+  include E with type exp = exp0
+end
+module rec LExprF : LExprF = LExpr(LExprF)
+let e3 =
+  LExprF.eval [] (`Add(`App(`Abs("x",`Mult(`Var"x",`Var"x")),`Num 2), `Num 5))
+[%%expect{|
+module LExprT :
+  sig
+    type 'a exp =
+        [ `Abs of string * 'b
+        | `Add of 'b * 'b
+        | `App of 'b * 'b
+        | `Mult of 'b * 'b
+        | `Num of int
+        | `Var of string ]
+      constraint 'a = < exp : 'b; .. >
+  end
+module type LExprS =
+  sig
+    type exp0 = private [> a LExprT.exp ]
+    and a = < exp : exp0 >
+    type exp = exp0
+    val eval : (string * exp) list -> exp -> exp
+  end
+module LExpr :
+  (E : LExprS) ->
+    sig
+      type exp = E.exp
+      type a = E.a
+      type exp0 = E.a LExprT.exp
+      module SLam :
+        sig
+          type exp0 = E.a LamT.exp
+          type exp = E.exp
+          type a = E.a
+          module LVar :
+            sig
+              type exp0 = VarT.exp
+              type exp = E.exp
+              val eval : (string * exp) list -> exp0 -> exp
+            end
+          val eval : (string * E.exp) list -> exp0 -> exp
+        end
+      module SExpr :
+        sig
+          type exp0 = E.a ExprT.exp
+          type exp = E.exp
+          type a = E.a
+          module LVar :
+            sig
+              type exp0 = VarT.exp
+              type exp = E.exp
+              val eval : (string * exp) list -> exp0 -> exp
+            end
+          val map : (E.exp0 -> E.exp0) -> exp0 -> exp
+          val eval : (string * LVar.exp) list -> exp0 -> LVar.exp
+        end
+      val eval : (string * SExpr.LVar.exp) list -> exp0 -> exp
+    end
+module type LExprF =
+  sig
+    type exp0 = a LExprT.exp
+    and a = < exp : exp0 >
+    type exp = exp0
+    val eval : (string * exp) list -> exp -> exp
+  end
+module rec LExprF : LExprF/2
```

Comment:
@Octachron, why is the signature printed as `LExprF/2` rather than `LExprF` here? I would expect the latter as module types and modules live in different namespaces, and there is only one module type named `LExprF` here I think.

Reply:
The issue seems to come from here :
https://github.com/ocaml/ocaml/blob/42adb2793657f4ef4f72eb9fc362c73e8eb2eb9f/typing/out_type.ml#L1810

Reply:
Indeed, the namespace switch is correct `Mty_alias` represents an alias to a specific module and it belongs to the module namespace. Moreover, at the outcometree level those aliases are printed as (module M_name):
```ocaml
module M: sig module L = List end = struct module L = struct end end
```
>        Modules do not match: sig end is not included in (module List)